Theory/CD.md

```diff
@@ -0,0 +1,446 @@
+- **Technical Debt**: If not managed properly, feature flags can become technical debt when old flags are not removed after they are no longer needed.
+
+**Example of Implementation**:  
+Popular tools like **LaunchDarkly**, **Feature Toggle** in **GitLab**, and **Unleash** offer robust platforms for managing feature toggles in production environments. Additionally, feature toggles can be implemented in-house using simple logic in code.
+
+---
+
+#### **5. Rollback Strategies**
+
+**Definition**:  
+A Rollback Strategy is a plan for reverting a system or application to a previous stable version when an issue is detected after deployment. It’s essential for maintaining uptime and minimizing the impact of issues on users.
+
+![My Image](../Images/rollback.png)
+
+**How It Works**:
+- **Version Control**: The system keeps track of previous versions of the application. When an issue is detected, the system or operations team can roll back to the last known good version.
+- **Rollback Process**:
+  1. Detect a failure or issue in the
+
+ current deployment.
+  2. Initiate a rollback to the previous version of the application or environment.
+  3. Monitor the rollback to ensure the system is restored to a stable state.
+
+**Example of Implementation**:  
+Kubernetes offers built-in rollback features with **kubectl rollout undo**, making it easy to revert to a previous deployment. Tools like **GitLab CI/CD** or **Jenkins** can also automate rollback in the event of a failed deployment.
+
+---
+
+#### **6. Continuous Monitoring**
+
+**Definition**:  
+Continuous Monitoring is the practice of constantly tracking the performance, stability, and health of the application in production. It ensures that any issues are detected as soon as they arise.
+
+![My Image](../Images/ContinousMoni.webp)
+Audit & Assurance Platform for monitoring.
+
+**How It Works**:
+- **Automated Tools**: Monitoring tools track various metrics such as uptime, response times, error rates, and server load. Alerts are triggered if any of these metrics fall outside predefined thresholds.
+- **Monitoring Process**:
+  1. Integrate monitoring tools such as **Prometheus**, **Grafana**, or **Datadog** to track the health of the system.
+  2. Set up dashboards and alerts for real-time visibility and notifications of any anomalies.
+  3. Use automated monitoring to continuously assess the application, especially after deployments.
+
+**Example of Implementation**:  
+In a Kubernetes environment, tools like **Prometheus** can monitor cluster health and application performance, while **Grafana** provides dashboards for real-time data visualization.
+
+---
+
+These deployment strategies provide various ways to manage risk, improve deployment speed, and ensure stability in modern software delivery. Each has its unique use cases, and choosing the right strategy depends on the goals of the deployment and the resources available for implementation.
+
+
+### 9. How are passwords securely stored?
```

Comment:
Why 9.?

Reply:
that is the 9 overall question per gitlab